hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java

```diff
@@ -243,6 +243,35 @@ Long countTargetByFilters(Collection<TargetUpdateStatus> status, String searchTe
             @NotNull TargetFilterQuery targetFilterQuery);
 
     /**
+     * Finds all targets for all the given parameter {@link TargetFilterQuery}
+     * and that don't have the specified assigned distribution set.
+     * and returns not the full target but {@link TargetIdName}.
+     *
+     * @param pageRequest
+     *            the pageRequest to enhance the query for paging and sorting
+     * @param targetFilterQuery
+     *            {@link TargetFilterQuery}
+     * @return the found {@link TargetIdName}s
+     */
+    @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_TARGET)
+    Page<Target> findAllTargetIdsByTargetFilterQueryAndNonDS(@NotNull Pageable pageRequest,
```

Comment:
methods is not returning `TargetId` so I think the method should be re-named to `findAllTargetsByTargetFilterQueryAndNonAssignedDS` and I would also call it with `AssignedDS` then it's readable in the method signature if it's an `installedDS` or `assignedDS`